spyder/plugins/plots/plugin.py

```diff
@@ -155,3 +186,415 @@ def apply_plugin_settings(self, options):
+            self.set_shellwidget_from_id(shellwidget_id)
+            return fig_browser
+
+    def remove_shellwidget_from_id(self, shellwidget_id):
+        # If shellwidget_id is not in self._shellwidgets, it simply means
+        # that shell was not a Python-based console (it was a terminal)
+        if shellwidget_id in self._shellwidgets:
+            fig_browser = self._shellwidgets.pop(shellwidget_id)
+            self.remove_widget(fig_browser)
+            fig_browser.close()
+
+    def set_shellwidget_from_id(self, shellwidget_id):
+        if shellwidget_id in self._shellwidgets:
+            fig_browser = self._shellwidgets[shellwidget_id]
+            fig_browser.setup(self._options)
+            self.set_current_widget(fig_browser)
+
+    # --- API
+    # ------------------------------------------------------------------------
+    def show_figure_menu(self, qpoint):
+        widget = self.current_widget()
+        if widget:
+            figcanvas = widget.figviewer.figcanvas
+            point = figcanvas.mapToGlobal(qpoint)
+            self.get_menu('context').popup(point)
+
+    def show_thumbnail_menu(self, qpoint):
+        pass
+
+    def mute_inline_plotting(self, value):
+        self.set_option('mute_inline_plotting', value)
+
+    def show_outline(self, value):
+        self.set_option('show_plot_outline', value)
+
+    def fit_to_window(self, value):
+        self.set_option('auto_fit_plotting', value)
+
+    def save_plot(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            widget.thumbnails_sb.save_current_figure_as()
+
+    def save_all_plots(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            widget.thumbnails_sb.save_all_figures_as()
+
+    def copy_image(self):
+        widget = self._stack.currentWidget()
+        if widget and widget.figviewer and widget.figviewer.figcanvas.fig:
+            widget.figviewer.figcanvas.copy_figure()
+
+    def remove_plot(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            widget.thumbnails_sb.remove_current_thumbnail()
+
+        self.update_actions()
+
+    def remove_all_plots(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            widget.thumbnails_sb.remove_all_thumbnails()
+
+        self.update_actions()
+
+    def previous_plot(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            widget.thumbnails_sb.go_previous_thumbnail()
+
+    def next_plot(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            widget.thumbnails_sb.go_next_thumbnail()
+
+    def zoom_in(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            widget.zoom_in()
+
+    def zoom_out(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            widget.zoom_out()
+
+
+class PlotsPlugin(SpyderDockablePlugin):
```

Comment:
@jitseniesen please look at how this `PlotsPlugin` looks like. it is using the new API.